Removed empty description from variable declaration

marceloalmeida opened the pull request:
* Trimmed first empty line

marceloalmeida committed:
Removed empty description from variable declaration

* Trimmed first empty line

marceloalmeida committed:
Replaced git sources since PR to upstream is stale

marceloalmeida committed:
Create release.yml

marceloalmeida committed:
Fixed documentation

marceloalmeida committed:
Merge branch 'master' of github.com:marceloalmeida/terraform-variables-generator

marceloalmeida committed:
Excluded darwin 386 from builds

marceloalmeida committed:
Create go.yml

alexandrst88 commented on cmd/cmd.go:
```diff
@@ -8,8 +8,8 @@ import (
 
-	"github.com/alexandrst88/terraform-variables-generator/pkg/generator"
-	"github.com/alexandrst88/terraform-variables-generator/pkg/utils"
+	"github.com/marceloalmeida/terraform-variables-generator/pkg/generator"
```
?

marceloalmeida replied:
It was to be able to perform `go get...` on multiple places and build using my changes, just reverted.

alexandrst88 commented on go.mod:
```diff
@@ -1,4 +1,4 @@
-module github.com/alexandrst88/terraform-variables-generator
+module github.com/marceloalmeida/terraform-variables-generator
```
?

alexandrst88 commented on main.go:
```diff
@@ -1,6 +1,6 @@
 package main
 
-import "github.com/alexandrst88/terraform-variables-generator/cmd"
+import "github.com/marceloalmeida/terraform-variables-generator/cmd"
```
?

marceloalmeida replied:
It was to be able to perform `go get...` on multiple places and build using my changes, just reverted.

alexandrst88 commented on pkg/generator/vars_test.go:
```diff
@@ -5,7 +5,7 @@ import (
 	"testing"
 
-	"github.com/alexandrst88/terraform-variables-generator/pkg/utils"
+	"github.com/marceloalmeida/terraform-variables-generator/pkg/utils"
```
?

marceloalmeida replied:
It was to be able to perform `go get...` on multiple places and build using my changes, just reverted.

alexandrst88 commented on pkg/generator/vars.go:
```diff
@@ -9,17 +9,19 @@ import (
-  description = ""
-}
-{{end}}`))
+var varTemplate = template.Must(template.New("var_file").Funcs(template.FuncMap{"sub": sub}).Parse(`{{- $length := len . -}}
```
what's the output will be?

marceloalmeida replied:
I've also updated the `README.md`, the output will look like the following snippet
```
variable "ami" {}

variable "instance_type" {}

variable "cidr" {}

variable "enable_dns_hostnames" {}

variable "enable_dns_support" {}

variable "name" {}
```

alexandrst88 reviewed (changes requested):
Hi! thanks for PR, there is already GH pr and release workflow. Could please answer on the comments?

marceloalmeida committed:
Reverted to maintainer repository

marceloalmeida committed:
Merge branch 'master' of github.com:marceloalmeida/terraform-variables-generator

marceloalmeida committed:
Merge branch 'master' into master

marceloalmeida commented:
> Hi! thanks for PR, there is already GH pr and release workflow. Could please answer on the comments?

Hi @alexandrst88, already did all the changes and replied to comments, can you please review?

alexandrst88 commented on .github/workflows/go.yml:
```diff
@@ -0,0 +1,39 @@
+name: Go
```
I've pushed already GitHub action of PR and release flow, could you please remove this one?

marceloalmeida replied:
Done

alexandrst88 commented on pkg/generator/vars.go:
```diff
@@ -16,11 +16,13 @@ var replacer *strings.Replacer
-}
-{{end}}`))
+var varTemplate = template.Must(template.New("var_file").Funcs(template.FuncMap{"sub": sub}).Parse(`{{- $length := len . -}}
+{{- range $i, $v := . -}}
```
there is no newline after rendering. 

Could you pleas back description?

marceloalmeida replied:
Added new line the rendering, also fixed the README.md

alexandrst88 replied:
sorry, but Readme and template are missing `description`

marceloalmeida replied:
It was on purpose, I've removed `description` since is not a mandatory field in the same way that `type` and `default` are. I'm using this tool to automatically generate `variables.tf`, when is needed to add `description` and other variables fields, for example when documenting modules, these are added manually

marceloalmeida committed:
Removed Github action, already in place

marceloalmeida committed:
Added new line at the end of generated file. Fixed documentation

marceloalmeida committed:
Merge branch 'master' of github.com:marceloalmeida/terraform-variables-generator

marceloalmeida commented:
@alexandrst88 can you please have a look?

alexandrst88 commented:
> @alexandrst88 can you please have a look?

Yes, I did and leave a requests change with comment:
```
sorry, but Readme and template are missing description
```

Thanks :)

marceloalmeida commented:
> > @alexandrst88 can you please have a look?
> 
> Yes, I did and leave a requests change with comment:
> 
> ```
> sorry, but Readme and template are missing description
> ```
> 
> Thanks :)

https://github.com/alexandrst88/terraform-variables-generator/pull/11/files#r544140310

It was on purpose, I've removed `description` since is not a mandatory field in the same way that `type` and `default` are. I'm using this tool to automatically generate `variables.tf`, when is needed to add `description` and other variables fields, for example when documenting modules, these are added manually

alexandrst88 commented:
> > > @alexandrst88 can you please have a look?
> > 
> > 
> > Yes, I did and leave a requests change with comment:
> > ```
> > sorry, but Readme and template are missing description
> > ```
> > 
> > 
> > Thanks :)
> 
> https://github.com/alexandrst88/terraform-variables-generator/pull/11/files#r544140310
> 
> It was on purpose, I've removed `description` since is not a mandatory field in the same way that `type` and `default` are. I'm using this tool to automatically generate `variables.tf`, when is needed to add `description` and other variables fields, for example when documenting modules, these are added manually

okay. but I still dont want to remove `description` field. could you please add cmd argument to make it optional or enabled by default?

marceloalmeida committed:
Added command flag to place or not description on variables

marceloalmeida commented:
> > > > @alexandrst88 can you please have a look?
> > > 
> > > 
> > > Yes, I did and leave a requests change with comment:
> > > ```
> > > sorry, but Readme and template are missing description
> > > ```
> > > 
> > > 
> > > Thanks :)
> > 
> > 
> > https://github.com/alexandrst88/terraform-variables-generator/pull/11/files#r544140310
> > It was on purpose, I've removed `description` since is not a mandatory field in the same way that `type` and `default` are. I'm using this tool to automatically generate `variables.tf`, when is needed to add `description` and other variables fields, for example when documenting modules, these are added manually
> 
> okay. but I still dont want to remove `description` field. could you please add cmd argument to make it optional or enabled by default?

Done, added a command argument (`true` by default) that places the description.

alexandrst88 reviewed (approved):
LGTM
